pkg/route/ingress/ingress.go

```diff
@@ -370,6 +372,28 @@ func (c *Controller) sync(key queueKey) error {
 		return err
 	}
 
+	// If the ingress specifies an ingressclass and the ingressclass does
+	// not specify openshift.io/ingress-to-route as its controller, ignore
+	// the ingress.
+	var ingressClassName *string
+	if v, ok := ingress.Annotations["kubernetes.io/ingress.class"]; ok {
+		ingressClassName = &v
+	} else {
+		ingressClassName = ingress.Spec.IngressClassName
+	}
+	if ingressClassName != nil {
+		ingressclass, err := c.ingressclassLister.Get(*ingressClassName)
+		if kerrors.IsNotFound(err) {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		if ingressclass.Spec.Controller != routev1.IngressToRouteIngressClassControllerName {
```

Comment:
Was this constant added before or after we started rebasing to k8s v1.21?

I'd rather not pull in go.mod changes that try to add k8s 1.21 deps in this PR. If this is the only reason that we need an openshift/api bump, can we hard-code the controller name here and mark a TODO to use the api constant?

Reply:
It was merged after.  https://github.com/openshift/api/commits/abd04f148867b326211c90f798235565572efd01